File: Data Structures & Algorithms/max-water-container/submission-0.rs

impl Solution {
    pub fn max_area(heights: Vec<i32>) -> i32
    {
        let (mut i, mut j) = (0, heights.len()-1);
        let mut max_area = 0;

        while i < j {
            let area = (j-i) as i32 * std::cmp::min(heights[i],heights[j]);
            max_area = max_area.max(area);

            if heights[i] < heights[j] {
                i += 1;
            } else { // yuck
                j -= 1;
            }
        }

        max_area
    }
}

File: Data Structures & Algorithms/longest-common-subsequence/submission-0.cpp

class Solution {
public:
    int longestCommonSubsequence(string text1, string text2)
    {
        auto const m = text1.size();
        auto const n = text2.size();

        auto dp = vector<vector<int>>(m,vector<int>(n));

        dp[0][0] = (text1[0] == text2[0]) ? 1 : 0;

        for (auto i = 1; i < m; ++i) {
            dp[i][0] = (text1[i] == text2[0]) ? 1 : dp[i-1][0];
        }
        for (auto j = 1; j < n; ++j) {
            dp[0][j] = (text1[0] == text2[j]) ? 1 : dp[0][j-1];
        }

        for (auto i = 1; i < m; ++i)
            for (auto j = 1; j < n; ++j)
        {
            dp[i][j] = (text1[i] == text2[j]) ? 1 + dp[i-1][j-1] : max(dp[i][j-1], dp[i-1][j]);
        }

        return dp[m-1][n-1];
    }
};
